sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -235,5 +205,11 @@ internal static string CreateRequestFailedMessageWithContent(Response response,
 
             return null;
         }
+
+        private class ErrorResponse
```

Comment:
nit: will this use reflection or would a JsonConverter be overkill?

Reply:
JsonSerializer uses reflection internally, but it's optimized and pretty fast we can optimize this code path more if we ever find it being too slow.